app/assets/javascripts/discourse/widgets/header.js

```diff
@@ -276,10 +276,51 @@ export function attachAdditionalPanel(name, toggle, transformAttrs) {
   additionalPanels.push({ name, toggle, transformAttrs });
 }
 
+let addedDismissHandler = false;
 export default createWidget("header", {
   tagName: "header.d-header.clearfix",
   buildKey: () => `header`,
 
+  init() {
+    // Allow first notification to be dismissed on a click anywhere
+    if (
+      this.currentUser &&
+      !this.currentUser.get("read_first_notification") &&
+      !this.currentUser.get("enforcedSecondFactor") &&
+      !addedDismissHandler
+    ) {
+      addedDismissHandler = true;
+      document.addEventListener(
```

Comment:
I don't love adding a click handler in a widget `init()`.

It might be easier to add the click handler in the header component `didInsertElement`. Then you could trigger an appEvent that this widget listens to.

Reply:
@eviltrout That's right, I forgot about app events for widgets - I've moved the addition of the event listener out to the site-header component :+1: 

Hopefully this looks a bit better!